File: app/src/main/java/com/commit451/gitlab/viewHolder/ProjectViewHolder.java
package com.commit451.gitlab.viewHolder;

import com.commit451.gitlab.R;
import com.commit451.gitlab.api.GitLabClient;
import com.commit451.gitlab.model.api.Project;
import com.github.ivbaranov.mli.MaterialLetterIcon;

import android.graphics.Color;
import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Projects, yay!
 * Created by Jawn on 6/11/2015.
 */
public class ProjectViewHolder extends RecyclerView.ViewHolder {

    public static ProjectViewHolder create(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_project, parent, false);
        return new ProjectViewHolder(view);
    }

    @Bind(R.id.project_image) ImageView mImageView;
    @Bind(R.id.project_letter) MaterialLetterIcon mLetterView;
    @Bind(R.id.project_title) TextView mTitleView;
    @Bind(R.id.project_description) TextView mDescriptionView;
    @Bind(R.id.project_stars) TextView mStarsView;
    @Bind(R.id.project_forks) TextView mForksView;
    @Bind(R.id.project_visibility) ImageView mVisibilityView;

    public ProjectViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    public void bind(Project project, int color) {
        if (project.getAvatarUrl() != null && !project.getAvatarUrl().equals(Uri.EMPTY)) {
            mLetterView.setVisibility(View.GONE);

            mImageView.setVisibility(View.VISIBLE);
            GitLabClient.getPicasso()
                    .load(project.getAvatarUrl())
                    .into(mImageView);
        } else {
            mImageView.setVisibility(View.GONE);

            mLetterView.setVisibility(View.VISIBLE);
            mLetterView.setLetter(project.getName().substring(0, 1));
            mLetterView.setLetterColor(Color.WHITE);
            mLetterView.setShapeColor(color);
        }

        mTitleView.setText(project.getNameWithNamespace());
        if (!TextUtils.isEmpty(project.getDescription())) {
            mDescriptionView.setVisibility(View.VISIBLE);
            mDescriptionView.setText(project.getDescription());
        } else {
            mDescriptionView.setVisibility(View.GONE);
            mDescriptionView.setText("");
        }

        mStarsView.setText(project.getStarCount() + "");
        mForksView.setText(project.getForksCount() + "");

        if (project.isPublic()) {
            mVisibilityView.setImageResource(R.drawable.ic_public_24dp);
        } else {
            mVisibilityView.setImageResource(R.drawable.ic_private_24dp);
        }
    }
}

File: app/src/main/java/com/commit451/gitlab/viewHolder/ProjectViewHolder.kt
package com.commit451.gitlab.viewHolder

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import coil.api.load
import coil.transform.CircleCropTransformation
import com.commit451.addendum.recyclerview.bindView
import com.commit451.gitlab.R
import com.commit451.gitlab.model.api.Project
import com.commit451.gitlab.model.api.VISIBILITY_INTERNAL
import com.commit451.gitlab.model.api.VISIBILITY_PUBLIC
import com.github.ivbaranov.mli.MaterialLetterIcon

/**
 * Projects, yay!
 */
class ProjectViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): ProjectViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_project, parent, false)
            return ProjectViewHolder(view)
        }
    }

    private val image: ImageView by bindView(R.id.project_image)
    private val iconLetter: MaterialLetterIcon by bindView(R.id.project_letter)
    private val textTitle: TextView by bindView(R.id.project_title)
    private val textDescription: TextView by bindView(R.id.project_description)
    private val iconVisibility: ImageView by bindView(R.id.project_visibility)

    fun bind(project: Project, color: Int) {
        // There is no longer a way that we can load images for private
        // repos, unfortunately. Passing the access token as a param
        // no longer works. We just have to show the letter for private
        // repos
        if (project.avatarUrl.isNullOrBlank() || project.visibility != VISIBILITY_PUBLIC) {
            image.visibility = View.GONE
            iconLetter.visibility = View.VISIBLE
            iconLetter.letter = project.name!!.substring(0, 1)
            iconLetter.letterColor = Color.WHITE
            iconLetter.shapeColor = color
        } else {
            iconLetter.visibility = View.GONE
            image.visibility = View.VISIBLE
            image.load(project.avatarUrl) {
                transformations(CircleCropTransformation())
            }
        }

        textTitle.text = project.nameWithNamespace
        if (!project.description.isNullOrEmpty()) {
            textDescription.visibility = View.VISIBLE
            textDescription.text = project.description
        } else {
            textDescription.visibility = View.GONE
            textDescription.text = ""
        }

        val visibilityResource = when (project.visibility) {
            VISIBILITY_PUBLIC -> R.drawable.ic_public_24dp
            VISIBILITY_INTERNAL -> R.drawable.ic_lock_open_24dp
            else -> R.drawable.ic_private_24dp
        }
        iconVisibility.setImageResource(visibilityResource)
    }
}
